File: backend/app/models/question.py
import datetime
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base


class Question(Base):
    __tablename__ = "questions"

    id = Column(Integer, primary_key=True, index=True)
    form_id = Column(Integer, ForeignKey("forms.id", ondelete="CASCADE"), nullable=False)
    type = Column(String, nullable=False)  # short_text, long_text, multiple_choice, dropdown, email, number, yes_no, rating
    title = Column(String, nullable=False)
    description = Column(String, nullable=True)
    required = Column(Boolean, default=False, nullable=False)
    order_index = Column(Integer, nullable=False, default=0)
    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.datetime.utcnow, onupdate=datetime.datetime.utcnow)

    form = relationship("Form", back_populates="questions")
    options = relationship(
        "QuestionOption",
        back_populates="question",
        cascade="all, delete-orphan",
        order_by="QuestionOption.order_index"
    )
    answers = relationship(
        "Answer",
        back_populates="question",
        cascade="all, delete-orphan"
    )


class QuestionOption(Base):
    __tablename__ = "question_options"

    id = Column(Integer, primary_key=True, index=True)
    question_id = Column(Integer, ForeignKey("questions.id", ondelete="CASCADE"), nullable=False)
    label = Column(String, nullable=False)
    value = Column(String, nullable=False)
    order_index = Column(Integer, nullable=False, default=0)

    question = relationship("Question", back_populates="options")

File: frontend/types/question.ts
export type QuestionType =
  | "short_text"
  | "long_text"
  | "multiple_choice"
  | "dropdown"
  | "email"
  | "number"
  | "yes_no"
  | "rating";

export interface QuestionOption {
  id?: number;
  label: string;
  value: string;
  order_index?: number;
}

export interface Question {
  id: number;
  form_id: number;
  type: QuestionType;
  title: string;
  description?: string | null;
  required: boolean;
  order_index: number;
  created_at: string;
  updated_at: string;
  options: QuestionOption[];
}

export interface QuestionCreatePayload {
  type: QuestionType;
  title?: string;
  description?: string;
  required?: boolean;
  order_index?: number;
  options?: { label: string; value: string; order_index?: number }[];
}

export interface QuestionUpdatePayload {
  title?: string;
  description?: string | null;
  required?: boolean;
  type?: QuestionType;
  order_index?: number;
  options?: { id?: number; label: string; value: string; order_index?: number }[];
}
